spdmlib/src/common/key_schedule.rs

```diff
@@ -47,19 +49,23 @@ impl SpdmKeySchedule {
 
     pub fn derive_master_secret(
         &self,
+        spdm_version: SpdmVersion,
         hash_algo: SpdmBaseHashAlgo,
         key: &[u8],
     ) -> Option<SpdmDigestStruct> {
         let buffer = &mut [0; MAX_SPDM_MESSAGE_BUFFER_SIZE];
         let bin_str0 = self.binconcat(
             hash_algo.get_size(),
-            SPDM_VERSION_VALUE,
+            if spdm_version == SpdmVersion::SpdmVersion12 {
+                SPDM_VERSION_VALUE12
+            } else {
+                SPDM_VERSION_VALUE11
+            },
```

Comment:
Can we use better way to resolve it?

My concern is that this code will be change in SPDM 1.3 and any further update.

Reply:
> Can we use better way to resolve it?
> 
> My concern is that this code will be change in SPDM 1.3 and any further update.

Improved. Please check the latest update.